src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -510,5 +527,22 @@ public T GetHeaderValue<T>(string key)
 
             return Header.Value<T>(key);
         }
+
+        /// <summary>
+        /// Tries to get the value corresponding to the provided key from the JWT header { key, 'value' }.
+        /// </summary>
+        /// <remarks>If the key has no corresponding value, returns false. Otherwise returns true. </remarks>   
+        public bool TryGetHeaderValue<T>(string key, out T value)
+        {
+            if (string.IsNullOrEmpty(key))
+            {
+                value = default(T);
+                return false;
+            }
+
+            var result = Header.TryGetValue(key, out var jTokenValue);
+            value = jTokenValue == null ? default(T) : jTokenValue.ToObject<T>();
```

Comment:
This will need to be in a try/catch as jTokenValue.ToObject<T> could throw.

Reply:
Done.